ios/FluentUI/Core/Theme/Tokens/ControlTokens.swift

```diff
@@ -6,7 +6,7 @@
 import Foundation
 
 /// Base class for all Fluent control tokenization.
-open class ControlTokens: NSObject {
-    var globalTokens: GlobalTokens = GlobalTokens.shared
-    var aliasTokens: AliasTokens = AliasTokens.shared
+public class ControlTokens: NSObject {
+    lazy var globalTokens: GlobalTokens = FluentTheme.shared.globalTokens
```

Comment:
What are pros/cons on the approaches for these references?
1. lazy var?
2. computed property (var)?
3. let (constant)?

Reply:
Great question!
### lazy var

**Pros:**
- Single source of truth for the variable
- Can be directly set in our `init`
- Is able to fetch values from `globalTokens` and `aliasTokens`, which are not yet available at `init` time
- Does not need to be overridden, and thus does not need to be `public` outside Fluent library

**Cons:**
- Cannot be overridden by a subclass (e.g. `OutlineCardNudgeTokens`)
- Cannot be set at `init` time by a subclass since they need to be lazy

### computed property (var)

**Pros:**
- Is able to fetch values from `globalTokens` and `aliasTokens`, which are not yet available at `init` time
- Can be overridden by a subclass (e.g. `OutlineCardNudgeTokens`)

**Cons:**
- Cannot store override values from clients without separate storage properties (greatly increasing binary size)

### let (constant)

**Pros:**
- Easy to read

**Cons:**
- Cannot be changed by client in `init`
- Cannot be overridden by a subclass (e.g. `OutlineCardNudgeTokens`)
- Cannot be lazy and thus cannot reference `globalTokens` or `aliasTokens`, which are not available at `init`